### PR TITLE
Fix Async Errors

### PR DESCRIPTION
**Pull request by bh2smith**
Here are some suggested changes that fix up all the lint format and type errors.

based on top of:

https://github.com/cowprotocol/dune-client/pull/31

### CODE REVIEW

File: dune_client/client_async.py [hunk 1]
```diff
@@ -26,14 +32,11 @@ class AsyncDuneClient(BaseDuneClient):
     An asynchronous interface for Dune API with a few convenience methods
     combining the use of endpoints (e.g. refresh)
     """
-    _session = None
+
+    # _session = None
```
Comment (bh2smith) [inline dune_client/client_async.py]: This should be removed (if you're ok with it) -- not just commented out.

**Comment by bh2smith** [comment]
As for the `base_url` issue I have made some progress but still not quite there.

This issue appears to contain some relevant content: 

https://github.com/aio-libs/aiohttp/issues/6647

**Comment by bh2smith** [comment]
I fixed all by one test... now the only failing test is 

`test_parameters_recognized` The response seems to be returning the query's initial parameters and not picking up the ones that are being set.

File: dune_client/client_async.py [hunk 2]
```diff
@@ -78,15 +81,15 @@ async def _handle_response(
...
             headers=self.default_headers(),
         )
         return await self._handle_response(response)
 
     async def _post(self, url: str, params: Any) -> Any:
         self.logger.debug(f"POST received input url={url}, params={params}")
-        response = await self._session.get(
-            url,
+        response = await self._session.post(
+            url=f"{self.API_PATH}{url}",
```
Comment (bh2smith) [inline dune_client/client_async.py:91]: This was a subtle issues (calling get instead of post inside post)

**Comment by bh2smith** [comment]
Most recent commit fixes all the issues. If you merge this I would say everything is ready (except that one method that belongs in Query class).

**Comment by eliseygusev** [comment]
great! thanks a lot for the support!

**Pull request merged**